enterprise/ha/src/test/java/org/neo4j/kernel/ha/cluster/SwitchToSlaveTest.java

```diff
@@ -0,0 +1,135 @@
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.neo4j.cluster.client.ClusterClient;
+import org.neo4j.cluster.member.ClusterMemberAvailability;
+import org.neo4j.graphdb.DependencyResolver;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.extension.KernelExtensionFactory;
+import org.neo4j.kernel.ha.BranchedDataException;
+import org.neo4j.kernel.ha.BranchedDataPolicy;
+import org.neo4j.kernel.ha.DelegateInvocationHandler;
+import org.neo4j.kernel.ha.HaXaDataSourceManager;
+import org.neo4j.kernel.ha.com.RequestContextFactory;
+import org.neo4j.kernel.ha.com.master.Master;
+import org.neo4j.kernel.ha.com.slave.MasterClient;
+import org.neo4j.kernel.ha.id.HaIdGeneratorFactory;
+import org.neo4j.kernel.impl.api.UpdateableSchemaState;
+import org.neo4j.kernel.impl.nioneo.xa.NeoStoreXaDataSource;
+import org.neo4j.kernel.logging.ConsoleLogger;
+import org.neo4j.kernel.logging.Logging;
+import org.neo4j.kernel.logging.SystemOutLogging;
+import org.neo4j.kernel.monitoring.Monitors;
+import org.neo4j.test.Mute;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SwitchToSlaveTest
+{
+    private SwitchToSlave switchToSlave;
+    private boolean methodCalled = false;
+
+    @Rule
+    public Mute mute = Mute.muteAll();
+
+    @Before
+    public void setUp()
+    {
+        // create a store with a wrong tx
+        ConsoleLogger console = mock( ConsoleLogger.class );
+        Config config = new Config();
+        DependencyResolver resolver = mock( DependencyResolver.class );
+        HaIdGeneratorFactory idGeneratorFactory = mock( HaIdGeneratorFactory.class );
+        Logging logging = new SystemOutLogging();
+        DelegateInvocationHandler<Master> masterDelegateHandler = mock( DelegateInvocationHandler.class );
+        ClusterMemberAvailability clusterMemberAvailability = mock( ClusterMemberAvailability.class );
+        ClusterClient clusterClient = mock( ClusterClient.class );
+        RequestContextFactory requestContextFactory = mock( RequestContextFactory.class );
+        UpdateableSchemaState updateableSchemaState = mock( UpdateableSchemaState.class );
+        Monitors monitors = mock( Monitors.class );
+        Iterable<KernelExtensionFactory<?>> kernelExtensions = mock( Iterable.class );
+
+        switchToSlave = new SwitchToSlave( console, config, resolver, idGeneratorFactory, logging,
+                masterDelegateHandler, clusterMemberAvailability, clusterClient, requestContextFactory,
+                updateableSchemaState, monitors, kernelExtensions )
+        {
+            @Override
+            void stopServicesAndHandleBranchedStore( BranchedDataPolicy branchPolicy ) throws Throwable
+            {
+                methodCalled = true;
+                return;
+            }
+        };
+    }
+
+    @Test( expected = BranchedDataException.class )
```

Comment:
Please don't use annotations for unit tests expecting exceptions, it is a terrible, terrible feature.